### PR TITLE
feature: User Enrolment Integration (w/o auth)

### PR DESCRIPTION
**Pull request by Jaylin0312**
## Overview
This PR addresses the UI part of frontend-backend integration for user service. For more project details please refer to Issue 28 [here](https://github.com/cs301-itsa/project-2023-24t1-g1-t3/issues/28#issue-1945411369).

## Features
- [x] C - Enrol new user
- [x] R - View all users
- [x] U - Update user details
- [x] D - Delete a specific user
- [x] Alert dialog for delete button
- [x] Toasters for action completion (create & update)

## To be addressed in future PR
- [ ] User's points balance 

## Testing
To carry out testing for this PR, please follow the steps below:

1. Ensure backend services (Docker) are up and running
2. Ensure `role` table is populated with `OWNER`, `MANAGER`, `ENGINEER`, and `PRODUCT_MANAGER`
3. Set up the project
```
yarn install
yarn dev
```

## Demo

https://github.com/cs301-itsa-ay2324/admin-ui-proxy/assets/68802506/9149ac22-1afb-4a89-9098-0b50a7e073e0

### CODE REVIEW

File: src/components/users-form.tsx [hunk 1]
```diff
@@ -35,18 +37,40 @@ const UsersFormSchema = z.object({
 
 export function UsersForm({
   defaultValues,
+  id,
 }: {
   defaultValues?: z.infer<typeof UsersFormSchema>
+  id: string
```
Comment (jeromeandrewong) [inline src/components/users-form.tsx]: should make id optional for when we are not editing form

File: src/components/deleteUserDialog.tsx
Comment (jeromeandrewong) [inline src/components/deleteUserDialog.tsx:1]: nit: can probably keep to this repo's component naming convention

File: src/components/table/user/user-table.tsx [hunk 2]
```diff
@@ -3,6 +3,7 @@ import { QueryClient, QueryClientProvider, useQuery } from "react-query"
 
 import { DataTable } from "../data-table"
 import { Columns } from "./columns"
+import UserRole from "@/../types/enums"
 
 const UserTable = () => {
   const queryClient = new QueryClient()
```
Comment (jeromeandrewong) [inline src/components/table/user/user-table.tsx]: can help to remove `QueryClientProvider` from this page and from `points.tsx` and wrap it in `_app.tsx` instead, multiple providers will cause problems for react query 
```suggestion
    <QueryClientProvider client={queryClient}>
        <Component {...pageProps} />
    <QueryClientProvider client={queryClient}>
```

**Review by jeromeandrewong — COMMENTED** [review]
thanks for the PR! Mainly looks good, Ive added some minor comment

also the build (excluding tests) is failing 🚧

File: package-lock.json
Comment (jeromeandrewong) [inline package-lock.json:1]: also if we are using yarn to manage our deps maybe we should stick to that, non-trivial but it's unlikely that the same dependency tree will be maintained in both files, and you don't want different behavior depending on whether the build was done using yarn or npm.

Comment (Jaylin0312) [reply package-lock.json:1]: omg you are right I must have accidentally ran npm 😫, will remove this

File: src/components/users-form.tsx (hunk 1 above)
Comment (Jaylin0312) [reply src/components/users-form.tsx]: Thanks for catching this! Addressed in https://github.com/cs301-itsa-ay2324/admin-ui-proxy/pull/5/commits/a8ec0eb3a363bb5890f19bf1a0954402ad5ba618

File: src/components/deleteUserDialog.tsx
Comment (Jaylin0312) [reply src/components/deleteUserDialog.tsx:1]: Done at https://github.com/cs301-itsa-ay2324/admin-ui-proxy/pull/5/commits/51af3f5a389e82e687b4aeeabceb597d35dd229a

File: src/components/table/user/user-table.tsx (hunk 2 above)
Comment (Jaylin0312) [reply src/components/table/user/user-table.tsx]: Alright, made the changes at https://github.com/cs301-itsa-ay2324/admin-ui-proxy/pull/5/commits/68211eaacd0eb7e5fe1469ef93fbdf286d980a4a

**Review by jeromeandrewong — APPROVED** [review]
LGTM 🚢

**Pull request merged**